docs/design/2020-09-12-utf8mb4-pinyin-order.md

```diff
@@ -0,0 +1,117 @@
+# Proposal: support `pinyin` order for `utf8mb4` charset
+
+- Author(s):     [xiongjiwei](https://github.com/xiongjiwei)
+- Last updated:  2020-09-20
+- Discussion at: https://github.com/pingcap/tidb/issues/19747
+
+## Abstract
+This proposal proposes a new feature that supports `pinyin` order for chinese character.
+
+## Background
+It's unable now to order by a column based on it's pinyin order. For example:
+
+```sql
+create table t(
+	a varchar(100)
+)
+charset = 'utf8mb4' collate = 'utf8mb4_zh_0900_as_cs';
+
+# insert some data:
+insert into t values ("中文"), ("啊中文");
+
+# a query requires to order by column a in its pinyin order:
+select * from t order by a;
++-----------+
+| a         |
++-----------+
+| 啊中文    |
+| 中文      |
++-----------+
+2 rows in set (0.00 sec)
+```
+
+## Proposal
+
+`pinyin` order for Chinese character supported by this proposal will add a new collation named `utf8mb4_general_zh_ci` which are exactly same with `gbk_chinese_ci`. Collation `utf8mb4_general_zh_ci` is for charset `utf8mb4`.
+
+Following SQL statements should have same result.
+```sql
+# order
+select * from t order by a collate utf8mb4_general_zh_ci;
+select * from t order by convert(a using gbk);
+select * from t order by convert(a using gbk) collate gbk_chinese_ci;
+
+# sort key
+select weight_string(a collate utf8mb4_general_zh_ci);
+select weight_string(convert(a using gbk));
```

Comment:
We don't have to make `weight_string()` 100% the same with GBK, after all, 'sort_key' is just something internal.

`utf8mb4_general_zh_ci` works if the order of characters are the same with `gbk`.